react-components/src/components/Architecture/FilterButton.tsx

```diff
@@ -101,61 +93,73 @@ export const FilterButton = ({
     return <></>;
   }
   return (
-    <Dropdown
+    <Menu
+      style={{
+        minWidth: '100px',
+        overflow: 'auto',
+        flexDirection
+      }}
+      floatingProps={{ middleware: [offset(TOOLBAR_HORIZONTAL_PANEL_OFFSET)] }}
       visible={isOpen}
       hideOnSelect={false}
+      onOpenChange={(open: boolean) => {
+        setOpen(open);
+      }}
       appendTo={'parent'}
-      placement={usedInSettings ? 'bottom-end' : 'auto-start'}
-      content={
-        <div ref={menuRef}>
-          <Menu
+      placement={usedInSettings ? 'auto-start' : 'bottom-end'}
+      disableCloseOnClickInside
+      renderTrigger={(props: any) => (
+        <CogsTooltip
+          content={<LabelWithShortcut label={label} command={command} />}
+          disabled={usedInSettings || label === undefined}
+          appendTo={document.body}
+          placement={placement}>
+          <Button
+            type={usedInSettings ? 'tertiary' : getButtonType(command)}
+            icon={
+              usedInSettings ? (
+                isOpen ? (
+                  <ChevronUpIcon />
+                ) : (
+                  <ChevronDownIcon />
+                )
+              ) : (
+                <IconComponent iconName={icon} />
+              )
+            }
+            key={uniqueId}
+            disabled={!isEnabled}
+            toggled={isOpen}
+            iconPlacement="right"
+            aria-label={command.getLabel(t)}
             style={{
-              minWidth: '100px',
-              overflow: 'auto',
-              flexDirection
+              minWidth: usedInSettings ? OPTION_MIN_WIDTH : undefined,
+              padding: usedInSettings ? DEFAULT_PADDING : undefined
+            }}
+            {...props}
+            onClick={(event: MouseEvent<HTMLElement>) => {
+              props.onClick(event);
+              event.stopPropagation();
+              event.preventDefault();
```

Comment:
Ideally all event propagation prevent should happen at the top, that before` props.onClick() `is called

Reply:
Hmm, okay :thinking: Is there a reason for that?